py/ig_compress_equal_clusters.py

```diff
@@ -39,49 +81,63 @@
 
     args = parser.parse_args()
 
-    if (not args.tmp_file):
-        args.tmp_file = tempfile.mkstemp(suffix=".fa", prefix="igrec_")[1]
-    if (not args.map_file):
-        args.map_file = tempfile.mkstemp(suffix=".map", prefix="igrec_")[1]
+    if (not args.tmp_fa_file):
+        args.tmp_fa_file = tempfile.mkstemp(suffix=".fa", prefix="igrec_")[1]
+    if (not args.tmp_rcm_file):
+        args.tmp_rcm_file = tempfile.mkstemp(suffix=".rcm", prefix="igrec_")[1]
 
     print "Command line: %s" % " ".join(sys.argv)
 
     home_directory = os.path.abspath(os.path.dirname(os.path.realpath(__file__))) + '/../'
     run_trie_compressor = os.path.join(home_directory, 'build/release/bin/./ig_trie_compressor')
-    cmd_line = "%s -i %s -o %s -m %s" % (run_trie_compressor, args.input, args.tmp_file, args.map_file)
+    cmd_line = "%s -i %s -o %s -m %s" % (run_trie_compressor, args.input, args.tmp_fa_file, args.tmp_rcm_file)
     os.system(cmd_line)
 
-    with open(args.map_file) as fin:
-        targets = [int(line.strip()) for line in fin]
+    with open(args.tmp_rcm_file) as fin:
+        input_read_num2compressed_cluster = [int(line.strip()) for line in fin]
 
+    input_read_num2mult = []
+    cluster2input_read_num = {}
     with open(args.input) as fin:
-        mults = [parse_abundance(record.description) for record in SeqIO.parse(fin, "fasta")]
+        for i, record in enumerate(SeqIO.parse(fin, "fasta")):
+            cluster, mult = parse_cluster_mult(record.description)
+            cluster2input_read_num[cluster] = i
+            input_read_num2mult.append(mult)
 
-    final_mults = [0] * (max(targets) + 1) if len(targets) else []
-    for target, mult in zip(targets, mults):
-        final_mults[target] += mult
+    compressed_cluster2mult = defaultdict(int)
+    for compressed_cluster, mult in zip(input_read_num2compressed_cluster, input_read_num2mult):
+        compressed_cluster2mult[compressed_cluster] += mult
 
-    with smart_open(args.tmp_file, "r") as fin, smart_open(args.output, "w") as fout:
+    # Fix IDs
+    with smart_open(args.tmp_fa_file, "r") as fin, smart_open(args.output, "w") as fout:
         for i, record in enumerate(SeqIO.parse(fin, "fasta")):
-            record.id = record.description = "cluster___%d___size___%d" % (i, final_mults[i])
+            record.id = record.description = "cluster___%d___size___%d" % (i, compressed_cluster2mult[i])
             SeqIO.write(record, fout, "fasta")
 
-
+    # Fix RCM if provided
     if args.rcm:
         print "Fixing RCM file..."
+        print "Check input consistency..."
+        if not check_fa_rcm_consistency(args.input, args.rcm):
+            exit(1)
+
         if not args.output_rcm:
             args.output_rcm = args.rcm
 
         with open(args.rcm, "r") as fin:
-            rcm = [line.strip().split("\t") for line in fin]
-            rcm = [(id, int(cluster)) for id, cluster in rcm]
+            _rcm = [line.strip().split("\t") for line in fin]
```

Comment:
Well, I think it was better before. Either come up with meaningful name or leave it rcm.